packages/ra-core/src/dataProvider/useCreate.ts

```diff
@@ -100,36 +100,36 @@ export const useCreate = <
                     meta: callTimeMeta,
                 })
                 .then(({ data }) => data),
-        {
-            ...options,
-            onSuccess: (
-                data: ResultRecordType,
-                variables: Partial<UseCreateMutateParams<RecordType>> = {},
-                context: unknown
-            ) => {
-                const { resource: callTimeResource = resource } = variables;
-                queryClient.setQueryData(
-                    [callTimeResource, 'getOne', { id: String(data.id) }],
-                    data
-                );
-                queryClient.invalidateQueries([callTimeResource, 'getList']);
-                queryClient.invalidateQueries([
-                    callTimeResource,
-                    'getInfiniteList',
-                ]);
-                queryClient.invalidateQueries([callTimeResource, 'getMany']);
-                queryClient.invalidateQueries([
-                    callTimeResource,
-                    'getManyReference',
-                ]);
+        ...options,
+        onSuccess: (
+            data: ResultRecordType,
+            variables: Partial<UseCreateMutateParams<RecordType>> = {},
+            context: unknown
+        ) => {
+            const { resource: callTimeResource = resource } = variables;
+            queryClient.setQueryData(
+                [callTimeResource, 'getOne', { id: String(data.id) }],
+                data
+            );
+            queryClient.invalidateQueries({
+                queryKey: [callTimeResource, 'getList'],
+            });
+            queryClient.invalidateQueries({
+                queryKey: [callTimeResource, 'getInfiniteList'],
+            });
+            queryClient.invalidateQueries({
+                queryKey: [callTimeResource, 'getMany'],
+            });
+            queryClient.invalidateQueries({
+                queryKey: [callTimeResource, 'getManyReference'],
+            });
 
-                if (options.onSuccess) {
-                    options.onSuccess(data, variables, context);
-                }
-                // call-time success callback is executed by react-query
-            },
-        }
-    );
+            if (options.onSuccess) {
+                options.onSuccess(data, variables, context);
+            }
+            // call-time success callback is executed by react-query
+        },
+    });
 
     const create = (
```

Comment:
you need to add calltime/runtime side effect discrimination

Reply:
Added